### PR TITLE
Fix server compare to check expiry last

### PR DESCRIPTION
**Pull request by dboslee**
Bug a I found while working on some heartbeat related things. I was adding an additional check at the bottom of `compareServers` that was never being reached, finally found this to be the problem. :smile:

### CODE REVIEW

**Comment by zmb3** [comment]
What's the impact of this bug? Backports required?

File: lib/services/server.go [hunk 1]
```diff
@@ -97,19 +97,19 @@ func compareServers(a, b types.Server) int {
...
 	}
-
 	if !cmp.Equal(a.GetApps(), b.GetApps()) {
 		return Different
 	}
 	if !cmp.Equal(a.GetKubernetesClusters(), b.GetKubernetesClusters()) {
 		return Different
 	}
+	// This check should always be last.
+	if !a.Expiry().Equal(b.Expiry()) {
```
Comment (jimbishopp) [inline lib/services/server.go:110]: Maybe we should explain why as well?
```suggestion
	// OnlyTimestampsDifferent check must be after all Different checks.
	if !a.Expiry().Equal(b.Expiry()) {
```

**Review by jimbishopp — COMMENTED** [review]
This looks like the correct fix. It wasn't obvious at first why it should be last. Maybe, while you're here, you could add a godoc to the function explaining that it returns Equal when all fields are equal, OnlyTimestampsDifferent when all fields except Expiry are equal, and Different otherwise. 

Please add some test coverage. A simple check that would've failed before this fix would be fine, e.g. version and expiry different.

**Comment by dboslee** [comment]
> What's the impact of this bug? Backports required?

I believe this currently impacts dynamic labeling for kubernetes agents. As well as dynamic labels for apps before they were moved from `types.Server` -> `types.AppServer`, not sure what version started using `types.AppServer`.

File: lib/services/server.go (hunk 1 above)
Comment (dboslee) [reply lib/services/server.go]: Yes that is much clearer updated here [78df4b1](https://github.com/gravitational/teleport/pull/10380/commits/78df4b18a03667b2240d5b975cb0a6bd86b60f81)

**Review by jimbishopp — APPROVED** [review]

**Comment by zmb3** [comment]
Yep, v7 and v8 backports are fine.

**Review by zmb3 — APPROVED** [review]

**Review by r0mant — APPROVED** [review]

**Pull request merged**